src/coreclr/jit/flowgraph.cpp

```diff
@@ -917,7 +917,7 @@ GenTreeCall* Compiler::fgGetSharedCCtor(CORINFO_CLASS_HANDLE cls)
         memset(&resolvedToken, 0, sizeof(resolvedToken));
         resolvedToken.hClass = cls;
 
-        return impReadyToRunHelperToTree(&resolvedToken, CORINFO_HELP_READYTORUN_STATIC_BASE, TYP_BYREF);
+        return impReadyToRunHelperToTree(&resolvedToken, CORINFO_HELP_READYTORUN_CCTOR_TRIGGER, TYP_BYREF);
```

Comment:
While you are here, can you change this to return `TYP_VOID`? If I'm not mistaken the effect of the `TYP_BYREF` here is that we are reporting random garbage value to the GC as a byref.